utils.js

```diff
@@ -1,11 +1,147 @@
 'use strict';
 
-var utils = {
+/**
+ * Uses abstract comparison (==) NOT strict comparison (===)
+ */
+function contains(list, item) {
+  for(var i = 0; i < list.length; i++) {
+    if (list[i] == item) {
+      return true;
+    }
+  }
+
+  return false;
+};
+
+function getphoneCountryConfig(phoneSettings, countryCode) {
+  var found;
+  var keys = Object.keys(phoneSettings);
+
+  keys.forEach(function(key) {
+    if (!found && phoneSettings[key].phoneCodes.country == countryCode) {
+      found = phoneSettings[key];
+    }
+  });
+
+  return found;
+}
+
+var fdpnUtils = {
+  contains: contains,
+
   isValidPhoneNumber: function(value) {
     return /^(00|\+)[0-9]{2,3}[\-\s0-9]{9,13}$/.test(value);
+  },
+
+  /**
+   * phone format: +<country_code>-<carrier>-<number>
+   */
+  parsePhone: function(phone, phoneSettings) {
+    var regexpPhone = /^\+([0-9]{2,3})(\-([0-9]{1,2}))?\-([0-9]{6,8})$/;
+    if (!phone || !regexpPhone.test(phone)) {
+      return {
+        fkCountry: '',
+        cellTokens: {
+          countryCode: '',
+          carrierCode: '',
+          number: ''
+        }
+      };
+    }
+
+    var mathes = phone.match(regexpPhone);
+    var countryCode = mathes[1];
+    var carrierCode = mathes[3] || '';
+    var number = mathes[4];
+    var phoneCountryConfig = getphoneCountryConfig(phoneSettings, countryCode);
+
+    return {
+      fkCountry: phoneCountryConfig.iso2Code,
+      cellTokens: {
+        countryCode: countryCode,
+        carrierCode: carrierCode,
+        number: number
+      }
+    };
+  },
+
+  validateCountry: function(countryId, phoneSettings) {
+    if (!countryId) {
+      return {empty: true, invalid: true};
```

Comment:
@odino I got the idea from angular: we have different ways of saying a field is valid and it maybe be that sometimes we show different messages if the field is empty or valid or has a specific validation problem.

This is why I'm always going to have an `invalid` field, and I also have other fields for specify "what" failed (if it make sense).